### ISSUE TITLE
[Bug]: System check on UI throws "Wrong number of arguments" Nextcloud Hub 9 (30.0.0.14).

### ISSUE THREAD

**Issue by cfiehe** [nextcloud/server#47997]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After upgrading to Nextcloud Hub 9 (30.0.0.14), the system check on the web UI throws an exception stating that the wrong number of arguments was given:

![grafik](https://github.com/user-attachments/assets/e9f1483d-eecf-439c-a4af-d2958532ebc8)

### Steps to reproduce

1. Login as admin
2. Set the language of your profile to "German".
3. Click on "Administrative settings".

### Expected behavior

No error is thrown during system check.

### Nextcloud Server version

master

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.mycompany.de",
            "nextcloud.lx.mycompany.de"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "proxy": "int-proxy.mycompany.de:3128",
        "proxyexclude": [
            "127.0.0.0\/8",
            "localhost",
            ".mycompany.de"
        ],
        "objectstore": {
            "class": "\\OC\\Files\\ObjectStore\\S3",
            "arguments": {
                "bucket": "loc-ed-nextcloud",
                "concurrency": 8,
                "uploadPartSize": 268435456,
                "autocreate": false,
                "verify_bucket_exists": false,
                "key": "***REMOVED SENSITIVE VALUE***",
                "secret": "***REMOVED SENSITIVE VALUE***",
                "hostname": "s3-sb.mycompany.de",
                "port": 443,
                "use_ssl": true,
                "use_path_style": true,
                "proxy": false
            }
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "30.0.0.14",
        "overwritehost": "nextcloud.mycompany.de",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/nextcloud.mycompany.de",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 86400,
        "session_keepalive": true,
        "auto_logout": true,
        "mail_smtpmode": "sendmail",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": 25,
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "filelocking.enabled": true,
        "htaccess.RewriteBase": "\/",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "user": "nextcloud",
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "theme": "",
        "loglevel": 2,
        "maintenance": false,
        "maintenance_window_start": 1,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "updater.release.channel": "stable"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 3.0.0
  - bruteforcesettings: 3.0.0
  - cloud_federation_api: 1.13.0
  - comments: 1.20.1
  - contactsinteraction: 1.11.0
  - dashboard: 7.10.0
  - dav: 1.31.1
  - federatedfilesharing: 1.20.0
  - federation: 1.20.0
  - files: 2.2.0
  - files_downloadlimit: 3.0.0
  - files_external: 1.22.0
  - files_pdfviewer: 3.0.0
  - files_reminders: 1.3.0
  - files_sharing: 1.22.0
  - files_trashbin: 1.20.1
  - files_versions: 1.23.0
  - firstrunwizard: 3.0.0
  - groupfolders: 18.0.1
  - impersonate: 1.17.0
  - logreader: 3.0.0
  - lookup_server_connector: 1.18.0
  - nextcloud_announcements: 2.0.0
  - notifications: 3.0.0
  - oauth2: 1.18.1
  - password_policy: 2.0.0
  - photos: 3.0.2
  - privacy: 2.0.0
  - provisioning_api: 1.20.0
  - recommendations: 3.0.0
  - related_resources: 1.5.0
  - serverinfo: 2.0.0
  - settings: 1.13.0
  - sharebymail: 1.20.0
  - support: 2.0.0
  - systemtags: 1.20.0
  - text: 4.1.0
  - theming: 2.5.0
  - twofactor_backupcodes: 1.19.0
  - updatenotification: 1.20.0
  - user_ldap: 1.21.0
  - user_status: 1.10.0
  - viewer: 3.0.0
  - webhook_listeners: 1.1.0-dev
  - workflowengine: 2.12.0
Disabled:
  - admin_audit: 1.20.0
  - circles: 30.0.0-dev (installed 30.0.0-dev)
  - encryption: 2.18.0
  - files_antivirus: 5.5.7 (installed 5.5.7)
  - files_retention: 1.19.0 (installed 1.19.0)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - previewgenerator: 5.6.0 (installed 5.6.0)
  - survey_client: 2.0.0 (installed 2.0.0)
  - suspicious_login: 8.0.0
  - twofactor_nextcloud_notification: 4.0.0
  - twofactor_totp: 12.0.0-dev
  - weather_status: 1.10.0 (installed 1.1.0)
```

### Nextcloud Logs

```shell
{
   "reqId":"7LiuStDp91ulri0xZxbe",
   "level":3,
   "time":"2024-09-15T06:37:06+00:00",
   "remoteAddr":"10.1.1.202",
   "user":"myuser",
   "app":"no app in context",
   "method":"GET",
   "url":"/settings/ajax/checksetup",
   "message":"Exception running check OCA\\User_LDAP\\SetupChecks\\LdapConnection: The arguments array must contain 2 items, 1 given",
   "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0",
   "version":"30.0.0.14",
   "exception":{
      "Exception":"ValueError",
      "Message":"The arguments array must contain 2 items, 1 given",
      "Code":0,
      "Trace":[
         {
            "file":"/var/www/html/nextcloud/lib/private/L10N/L10NString.php",
            "line":68,
            "function":"vsprintf"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/L10N/L10N.php",
            "line":107,
            "function":"__toString",
            "class":"OC\\L10N\\L10NString",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/L10N/LazyL10N.php",
            "line":38,
            "function":"n",
            "class":"OC\\L10N\\L10N",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/apps/user_ldap/lib/SetupChecks/LdapConnection.php",
            "line":87,
            "function":"n",
            "class":"OC\\L10N\\LazyL10N",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/SetupCheck/SetupCheckManager.php",
            "line":34,
            "function":"run",
            "class":"OCA\\User_LDAP\\SetupChecks\\LdapConnection",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/apps/settings/lib/Controller/CheckSetupController.php",
            "line":147,
            "function":"runAll",
            "class":"OC\\SetupCheck\\SetupCheckManager",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
            "line":208,
            "function":"check",
            "class":"OCA\\Settings\\Controller\\CheckSetupController",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
            "line":114,
            "function":"executeController",
            "class":"OC\\AppFramework\\Http\\Dispatcher",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php",
            "line":161,
            "function":"dispatch",
            "class":"OC\\AppFramework\\Http\\Dispatcher",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/lib/private/Route/Router.php",
            "line":302,
            "function":"main",
            "class":"OC\\AppFramework\\App",
            "type":"::"
         },
         {
            "file":"/var/www/html/nextcloud/lib/base.php",
            "line":1001,
            "function":"match",
            "class":"OC\\Route\\Router",
            "type":"->"
         },
         {
            "file":"/var/www/html/nextcloud/index.php",
            "line":24,
            "function":"handleRequest",
            "class":"OC",
            "type":"::"
         }
      ],
      "File":"/var/www/html/nextcloud/lib/private/L10N/L10NString.php",
      "Line":68,
      "message":"Exception running check OCA\\User_LDAP\\SetupChecks\\LdapConnection: The arguments array must contain 2 items, 1 given",
      "exception":[
         
      ],
      "CustomMessage":"Exception running check OCA\\User_LDAP\\SetupChecks\\LdapConnection: The arguments array must contain 2 items, 1 given"
   },
   "id":"66e6809848750"
}
```

**Comment by gelbphoenix** [comment]
My LCMP¹ stack install throws the same error after installing Nextcloud Hub 9.

¹(Linux, Caddy, MariaDB, PHP)

**Comment by come-nc** [comment]
It seems since https://github.com/nextcloud/server/pull/32155 it’s not possible anymore to use the plural system without having %n in the translated string.
So either we fix that or we add %n in the string or we use an if instead.

**Comment by come-nc** [comment]
Fixed by https://github.com/nextcloud/server/pull/48028

**Comment by stanthewizzard** [comment]
I have the exact same issue

**Comment by gelbphoenix** [comment]
> I have the exact same issue

Should be resolved with Nextcloud 31 or (in a hacky and not recommended way) you could implement the fix #48028  in the LdapConnection.php yourself.

**Comment by stanthewizzard** [comment]
I will wait as it is in a production docker. 
Thanks

**Comment by joshtrichards** [comment]
> Should be resolved with Nextcloud 31

The fix is also in 30.0.1. It was backported.
